Dev

ohrytsiuk opened the pull request:
Implement project sceleton

ohrytsiuk committed:
Implement project sceleton

ohrytsiuk committed:
Remove unittest from requirements.pip because error

denald commented on .gitignore:
```diff
@@ -0,0 +1,104 @@
+*.mo
+*.pot
+
+# Django stuff:
```
We don't use Django, I guess we can remove it from here

denald commented on .gitignore:
```diff
@@ -0,0 +1,104 @@
+local_settings.py
+db.sqlite3
+
+# Flask stuff:
```
Same as above - we don't use flask

denald commented on .gitignore:
```diff
@@ -0,0 +1,104 @@
+/site
+
+# mypy
+.mypy_cache/
```
Consider cleaning up this file from unnecessary stuff

denald commented on README.md:
```diff
@@ -1 +1,31 @@
+Run `pip install pycodestyle` to perform project code inspection using `pep8`.
+ 
+## Run tests
+- To run unit tests execute `python unit.py`.
```
We will use `nose` as test runner, the command should be different

denald commented on README.md:
```diff
@@ -1 +1,31 @@
+### Prerequisites 
+- Python [3.4 or higher](https://www.python.org/downloads/)
+- Virtual environment [Virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/)
+- Cloned repository [git](git@github.com:repo/repo.git)
```
Wong git link

denald commented on README.md:
```diff
@@ -1 +1,31 @@
+- Docker-compose [install](https://docs.docker.com/compose/install/)
+
+### Setting up the project
+- create new virtual env `mkvirtualenv -p <path/to/python3> project_name`, 
```
Maybe it's better to name virtual env to something like `api-tests`

denald commented on README.md:
```diff
@@ -1 +1,31 @@
+- activate the env by running `workon project_name`
+- Install modules `pip install -r ~/<path to project>/requirements.pip`
+- Clone repository [rest_Spring_Docker](https://github.com/momel/rest_Spring_Docker)  
+- Up clone image: `cd ~/<path to your cloned repository>` and `docker-compose up -d`
```
It's not clear what we actually doing with this command, consider changing wording

denald commented on __init__.py:
```diff
@@ -0,0 +1,20 @@
+        raise RuntimeError('Test error!')
+
+
+if __name__ == '__main__':
```
We will run our tests with `nose` test runner, please remove these lines of code from here

denald commented on __init__.py:
```diff
@@ -0,0 +1,20 @@
+import unittest
```
This file should be moved to `functional` folder

denald replied:
Sorry, to `tests` folder

JonnyJoypad commented on .gitignore:
```diff
@@ -0,0 +1,104 @@
+*$py.class
+
+# C extensions
+*.so
```
Unnecessary stuff for us now:  C# or Shared objects (.so)

ohrytsiuk committed:
add to constants InitUsers class
fix imports in unit/__init__.py and functional/__init__.py
move __init__.py from root to tests
cleaned up file .gitignore
fixed link, virtualenv instruction, docker instruction, running instruction and added new lib on git on file readme.md
add version for modules in requirements.pip

ohrytsiuk committed:
removed non actually test strings

ohrytsiuk commented:
> This one was right?
> "- create new virtual env `mkvirtualenv -p <path/to/python3> project_name`"
from readme.md
mkvirtualenv api-tests -p <path/to/python3>
for example: 
mkvirtualenv api-tests -p /usr/bin/python3
made new virtualenv api-tests

zzee19 committed:
add new file to constants __init__.py
changed name class UserConstants to DefaultUser and add new clases BaseUrl, Endpoints in file constants.py
changed functionality in file functional/__init__.py
added new file tests_login.py in functional
added new python package utils with files __init__ and helper.py
empty tests/__init__.py
change nose run instruction

ohrytsiuk committed:
add class InitInvalidUsers to constants.py
remove tests_login.py to the new branch
pycodestyle fix in helper.py
add configuration file .pylintrc
ad pylint in requirements.pip

ykobuk committed:
saving work progress

ykobuk committed:
returned to previous stage

abertash committed:
Add new file for test cool down time changes.
Add new function at init file for test cool down time changes.

ykobuk committed:
just saving work

abertash committed:
My propose for changes at constans.py: in the class "DefaultUser" I separate user and admin. In my tests I need login like user and like admin, and I think that it is best way.
ATTENTION! You will need refactor your code, because now DefaultUser.user != admin.

Add new function to get current cool down time.

End write tests for /cooldowntime

abertash committed:
func/init:
- Add new functions to get/set token life time.
- Add new function to get reset.

other:
- Add new tests for cool down time (positive, negative, boundary)
- Add new file with all tests for token life
- refactoring my own files

pbezrukov committed:
The first attempt of test

Shtepsel committed:
test of locking users completed

JonnyJoypad committed:
added 3 positive test, and 1 test with bug

Shtepsel committed:
changed locked_user in constants to locked_user= "/locked/user/" and item = "/item/"

abertash committed:
func/init:
- Add new functions to get all users.
- Change names of my own params, that must be understandable

other:
- Add new file with all tests for /users
- refactoring my own files

ohrytsiuk committed:
add new class tearDown in functional/__init__.py for close request session and reset API and python3 style code fix
add new file tests_admins for testing API /admin
add new file login for testing API /login
add new file logout for testing API /logout

pbezrukov committed:
Add new tests, methods of item

JonnyJoypad committed:
edited previous files, addednew tests for creating user and removing

Shtepsel committed:
saving work

Shtepsel committed:
saving work

Shtepsel committed:
saving work

JonnyJoypad committed:
edited remove user

JonnyJoypad committed:
added new tests (change pass and change pass with invalid data)

JonnyJoypad committed:
just rename file

pbezrukov committed:
tests for all items method, changes constants file, add to __init__.py

ykobuk committed:
just saving work

ykobuk committed:
just saving work

ykobuk committed:
just saving work

pbezrukov committed:
Refactoring tests

pbezrukov committed:
Change file tests_get_all_items_user, and file Constants(removed admin from users)

pbezrukov committed:
Some changes in the tests files

JonnyJoypad committed:
added new user tests, edited init file

ohrytsiuk committed:
refactoring code
pylint and pycodestyle fixes

JonnyJoypad committed:
added test for getting user name

pbezrukov committed:
Refactoring code

JonnyJoypad committed:
edited previous and added new tests to removing user

JonnyJoypad committed:
added new tests for deleting users

ykobuk committed:
just saving work

ykobuk committed:
just saving work

abertash committed:
change constants that other users will not have problems with renames.
refactoring.

abertash committed:
change constants that other users will not have problems with renames.
refactoring.

Commit:
added get_new_user to __init__ file, edited tests_get_user_name

ykobuk committed:
just saving work

ykobuk committed:
just saving work

Shtepsel committed:
just saving work

Commit:
pylint edit

JonnyJoypad committed:
All edited with pylint on a board

Shtepsel committed:
icnreasing test coverage

ohrytsiuk committed:
Merge pull request #18 from Social-projects-Rivne/ohrytsiuk

Ohrytsiuk

abertash committed:
Merge branch 'dev' into pbezrukov

abertash committed:
Merge pull request #19 from Social-projects-Rivne/pbezrukov

Pbezrukov

JonnyJoypad committed:
Added wrong passwords to constants, Tests_user_change_pass totally reworked

JonnyJoypad committed:
total refactoring of tests, added new constants, added 2 tests for changing pass

Shtepsel committed:
half of tests reworked

abertash committed:
refactoring.

abertash committed:
Merge branch 'dev' into abertash_tests

abertash committed:
Merge pull request #21 from Social-projects-Rivne/abertash_tests

abertash tests

Shtepsel committed:
second half of tests reworked

Shtepsel committed:
saving regress

Shtepsel committed:
test reworked with correspondance to dev/constants and dev/functional/init.py

Shtepsel committed:
Merge branch 'dev' into staff
fixed merge conflicts

Shtepsel committed:
Merge pull request #23 from Social-projects-Rivne/staff

Tests of locking users and smth else

Commit:
added @ddt to repetable tests

Commit:
pyling

ohrytsiuk committed:
Merge branch 'dev' into heneralchuk

pbezrukov committed:
made the file application when created class Wrapper for all method with __init__ method

JonnyJoypad committed:
Merge pull request #22 from Social-projects-Rivne/heneralchuk

Heneralchuk, user tests

pbezrukov committed:
the new file application, changes the file __init__ of the functional, for example look file tests_login. It is the new architecture, when logic exists separately tests

pbezrukov committed:
refactoring code

pbezrukov committed:
Delete application.py

ohrytsiuk committed:
refactoring code for new architecture
replace the api functions from functional/__init__.py to the new file application/application.py

pbezrukov committed:
reactoring

pbezrukov committed:
Merge pull request #24 from Social-projects-Rivne/pbezrukov

Implement ApiWrapper

ohrytsiuk committed:
Merge branch 'dev' into ohrytsiuk

ohrytsiuk committed:
Merge pull request #25 from Social-projects-Rivne/ohrytsiuk

refactoring code for new architecture

Commit:
Application changes in tests

JonnyJoypad committed:
Merge pull request #26 from Social-projects-Rivne/heneralchuk

Application changes in tests

Shtepsel committed:
reworked docstrins

Shtepsel committed:
docstings reworked

Shtepsel committed:
Merge branch 'doc_staff' of https://github.com/Social-projects-Rivne/RestPyTAC into doc_staff

# Conflicts:
#	tests/functional/tests_locked.py

Commit:
ref.

ohrytsiuk committed:
fixing docstrings
add tests for login, logout, admins with empty token

ohrytsiuk committed:
add ddt for some tests
docstring fixes

ohrytsiuk committed:
Merge pull request #27 from Social-projects-Rivne/ohrytsiuk

ohrytsiuk->dev

pbezrukov committed:
refactoring docstrings

pbezrukov committed:
Merge pull request #28 from Social-projects-Rivne/pbezrukov

refactoring docstrings

Shtepsel committed:
docstring in cooldown time test  updated

JonnyJoypad committed:
changed some asserts in tests, edoted doc strings

Shtepsel committed:
Merge pull request #30 from Social-projects-Rivne/doc_staff

doc-strings in tests_cooldowntime and tests_locked updated

JonnyJoypad committed:
Merge pull request #31 from Social-projects-Rivne/heneralchuk

edited doc strings
